src/encoder.rs

```diff
@@ -232,6 +232,9 @@ impl<W: Write> Encoder<W> {
 
     fn write_encoded_image_block(writer: &mut W, data_with_min_code_size: &[u8]) -> Result<(), EncodingError> {
         let (&min_code_size, data) = data_with_min_code_size.split_first().unwrap_or((&2, &[]));
+        if min_code_size > 11 || min_code_size < 2 {
```

Comment:
Most of the code changes are small, but this one could perhaps use a bit of explanation...

Reply:
It rejects invalid input. The method expects a code size valid for GIF followed by LZW data.